src/Numeric/Units/Dimensional/NonSI.hs

```diff
@@ -108,59 +118,296 @@ gee = mkUnitQ (ucumMetric "[g]" "g" "gee") 9.80665 $ meter / second ^ pos2
 Some US customary (that is, inch-pound) units.
 -}
 
-inch, foot, mil :: Fractional a => Unit 'NonMetric DLength a
-inch = mkUnitQ (ucum "[in_i]" "in" "inch") 2.54 $ centi meter
-foot = mkUnitQ (ucum "[ft_i]" "ft" "foot") 12 $ inch     -- 0.3048 m
-mil  = mkUnitQ (ucum "[mil_i]" "mil" "mil") 0.001 $ inch
-poundMass, ounce :: Fractional a => Unit 'NonMetric DMass a
-poundMass = mkUnitQ (ucum "[lb_av]" "lb" "pound") 0.45359237 $ kilo gram
-ounce     = mkUnitQ (ucum "[oz_av]" "oz" "ounce") (1 Prelude./ 16) $ poundMass
+-- | One international foot is one third of an international 'yard'.
```

Comment:
Agreed that `prop>` is much nicer. Were you able to check the 100 iterations thing? I have hade to Otherwise I will try with `Debug.Trace`. I've had to use, e.g., 
https://github.com/bjornbm/astro/blob/265916aa35b0269702e15eed879ec211d3cd4517/test/Astro/TestCelestrak.hs#L16 in the past:

    onceCheck = quickCheckWith stdArgs { maxSuccess = 1 }

Either way the checks here aren't very expensive.